redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/DBStackUpdater.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.redbeams.service.stack;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.responses.SslCertificateType;
+import com.sequenceiq.redbeams.configuration.DatabaseServerSslCertificateConfig;
+import com.sequenceiq.redbeams.configuration.SslCertificateEntry;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.domain.stack.SslConfig;
+
+@Service
+public class DBStackUpdater {
+
+    @Inject
+    private DatabaseServerSslCertificateConfig databaseServerSslCertificateConfig;
+
+    @Inject
+    private DBStackService dbStackService;
+
+    public void updateSslConfig(long id) {
+        Optional<DBStack> dbStackOpt = dbStackService.findById(id);
+        dbStackOpt.ifPresent(dbStack -> {
+            SslConfig sslConfig = dbStack.getSslConfig();
+            String cloudPlatform = dbStack.getCloudPlatform();
+            String region = dbStack.getRegion();
+            Set<SslCertificateEntry> allCertificates = databaseServerSslCertificateConfig.getCertsByCloudPlatformAndRegion(cloudPlatform, region);
+            sslConfig.setSslCertificates(allCertificates.stream().map(SslCertificateEntry::getCertPem).collect(Collectors.toSet()));
+            int activeVersion = databaseServerSslCertificateConfig.getMaxVersionByCloudPlatformAndRegion(cloudPlatform, region);
+            SslCertificateEntry activeSslCert = allCertificates.stream()
+                    .filter(sslCert -> sslCert.getVersion() == activeVersion)
+                    .findFirst()
+                    .orElseThrow(NotFoundException.notFound(String.format("Active SSL cert cannot be found for %s", dbStack.getName())));
+            sslConfig.setSslCertificateActiveVersion(activeVersion);
+            sslConfig.setSslCertificateActiveCloudProviderIdentifier(activeSslCert.getCloudProviderIdentifier());
+            sslConfig.setSslCertificateType(SslCertificateType.CLOUD_PROVIDER_OWNED);
+            dbStackService.save(dbStack);
```

Comment:
This logic must not be executed if the DB stack does not have SSL enabled. In particular, we shall check for `sslConfig.getSslCertificateType() == SslCertificateType.CLOUD_PROVIDER_OWNED`, and its setter need not be invoked explicitly. (As I see, the flow step always gets executed, so is the handler, and the handler does not have any conditional checks for this either.)

Reply:
I have merged this and the above comments and I have added a condition and the logic will be executed if SSLConfig is not null and the type is CLOUD_PROVIDER_OWNED